README.md
# HTML5_Framework
Framework for making games with pixi.js


changelog.md
# iVantage CHANGELOG r1:r135


## misc

### default
  
- [135](#135) - bumped to v3.0.6
  
- [134](#134) - added updates from bing blocks
  
- [133](#133) - renamed runloop - UpdateLoop.js
  
- [132](#132) - commenting
  
- [131](#131) - fixed audio fade in bug
  
- [130](#130) - tweaks to cropsprite, fixed movieclip events
  
- [129](#129) - fixed bootstrap variable scope error
  
- [128](#128) - top edge for game object doesn&#x27;t use frame
  
- [127](#127) - fixed ios9 resolution error
  
- [126](#126) - fixed ios9 resolution error
  
- [125](#125) - don&#x27;t unload for reload screen
  
- [124](#124) - SndSprite mute group fix
  
- [123](#123) - bootstrap atlas load resolution fix
  
- [122](#122) - corrected bootstrap scope error
  
- [121](#121) - updated sound manager config handling
  
- [120](#120) - corrected toggle
  
- [119](#119) - corrected toggle
  
- [118](#118) - corrected toggle
  
- [116](#116) - changed GameObject updateFrame to handle transforms
  
- [115](#115) - added Save utility
  
- [114](#114) - tween fade out fix
  
- [113](#113) - fixed chrome audio loop bug
  
- [112](#112) - added libs to source files.json
  
- [110](#110) - bumped to v3.0.5
  
- [108](#108) - bumped to v3.0.5
  
- [107](#107) - added crop sprite and camera cull
  
- [106](#106) - fixed empty manifest loading bug
  
- [105](#105) - corrected MovieClip to use texture =
  
- [103](#103) - added originalTouch pixi mod
  
- [102](#102) - fixed viewport resize
  
- [101](#101) - fixed viewport resize
  
- [100](#100) - added more debug to GameObject
  
- [99](#99) - added some extra bits to screen manager from dark ascent
  
- [98](#98) - removed resolution from gameobject updateFrame
  
- [97](#97) - fixed button pivot
  
- [96](#96) - updated pixi dragonbones
  
- [95](#95) - ok
  
- [93](#93) - bumped to v3.0.4
  
- [92](#92) - bumped to v3.0.3
  
- [91](#91) - bumped to v3.0.1
  
- [90](#90) - bumped to v3.0.4
  
- [89](#89) - bumped to v3.0.3
  
- [88](#88) - bumped to v3.0.3
  
- [87](#87) - bumped to v3.0.2
  
- [86](#86) - bumped to v3.0.1
  
- [85](#85) - added audio sprite task
  
- [84](#84) - fixed game object
  
- [83](#83) - added texturepacker grunt task
  
- [82](#82) - removed unused spine utils!
  
- [81](#81) - updated spine example
  
- [80](#80) - tweaks
  
- [79](#79) - turned Game into self invoking function
  
- [78](#78) - added GameObject class
  
- [76](#76) - clean
  
- [75](#75) - changed version number to string
  
- [74](#74) - renamed dragonbones file for clarity
  
- [73](#73) - adding grunt bump to template
  
- [70](#70) - bumped to v3.0.1
  
- [69](#69) - grunt bump is working
  
- [68](#68) - cleaning up screen manager
  
- [66](#66) - fixed image onerror issue
  
- [65](#65) - example particles follow mouse
  
- [64](#64) - changelog is working
  
- [53](#53) - trying to get changelog working
  
- [50](#50) - game template is all builder based now
  
- [49](#49) - adding scenes and builders to game template
  
- [48](#48) - tightening up the template
  
- [47](#47) - fixed bitmap font
  
- [46](#46) - drag and drop is working
  
- [45](#45) - working on drag and drop
  
- [44](#44) - adding drag n drop screen
  
- [43](#43) - added particles
  
- [42](#42) - working on bitmap fonts
  
- [41](#41) - added movieclip
  
- [40](#40) - renamed statics to settings
  
- [39](#39) - corrected spine update
  
- [38](#38) - added spine sprite wrapper
  
- [37](#37) - spine is working
  
- [36](#36) - spine assets displaying, need to get it animating!
  
- [35](#35) - added deploy and grunt to game template
  
- [34](#34) - souped up game template
  
- [33](#33) - getting spine working
  
- [32](#32) - added spine screen
  
- [31](#31) - updated todo
  
- [30](#30) - dragonbones works. Atlas loading is fixed
  
- [29](#29) - dragonbones example is nearly working
  
- [28](#28) - setting up dragonbones
  
- [27](#27) - started examples
  
- [26](#26) - adding text
  
- [25](#25) - some linting
  
- [24](#24) - loading classes refactored with init config
  
- [23](#23) - refactored image load state
  
- [22](#22) - more loading classes renamed
  
- [21](#21) - renamed load manifest
  
- [20](#20) - renamed pause controller
  
- [19](#19) - renamed pixistage to just stage
  
- [18](#18) - fixing loader
  
- [17](#17) - fixed button
  
- [16](#16) - renamed game_shell property to stage
  
- [15](#15) - adding trimming support
  
- [14](#14) - fixed resolution
  
- [13](#13) - found that webgl resolution is not working
  
- [12](#12) - trying to add auto changelogs
  
- [11](#11) - fixing resolution handling, currently still broken!
  
- [10](#10) - fixed pixi errors.
  
- [9](#9) - foo
  
- [8](#8) - it&#x27;s runnable!
  
- [7](#7) - fixed ajaxrequest
  
- [6](#6) - getting v3.0.0 working
  
- [5](#5) - adding game template
  
- [4](#4) - added bat file shortcut
  
- [3](#3) - refactored generic game
  
- [2](#2) - fixed grunt
  
- [1](#1) - initial file add
  




src/utils/Key.js
/**
 *
 *
 */
game_shell.Keys = {
	_pressed: {},
    //
	LEFT_KEY: 37,
    ENTER_KEY: 13,
	UP_KEY: 38,
	RIGHT_KEY: 39,
	DOWN_KEY: 40,
	SPACE_KEY: 32,
	SHIFT_KEY: 16,
	//key states
	UP:0,
	JUST_PRESSED:1,
	DOWN:2,
	onKeydown: function(event) {
		if(!this._pressed[event.keyCode]) this._pressed[event.keyCode] = this.JUST_PRESSED;
        var evt = event || window.event;
        if (evt.stopPropagation !== undefined) {
            evt.stopPropagation();
        } else {
            evt.cancelBubble = true;
        }
        if(evt.preventDefault) evt.preventDefault();
	},
	onKeyup: function(event) {
		delete this._pressed[event.keyCode];
        var evt = event || window.event;
        if (evt.stopPropagation !== undefined) {
            evt.stopPropagation();
        } else {
            evt.cancelBubble = true;
        }
        if(evt.preventDefault) evt.preventDefault();
    },
    update:function(dt){
    	for(var s in this._pressed){
            if(this._pressed.hasOwnProperty(s)){
    		  if(this._pressed[s] === this.JUST_PRESSED) this._pressed[s] = this.DOWN;
            }
    	}
    },
    clear:function(){
        for(var s in this._pressed){
            if(this._pressed.hasOwnProperty(s)){
                delete this._pressed[s];
            }
        }
    },
    enable: function(bool){
        if(!this.boundKeyUp) this.boundKeyUp = this.onKeyup.bind(this);
        if(!this.boundKeyDown) this.boundKeyDown = this.onKeydown.bind(this);
        //support iframes!
        var isIframe = window.parent !== window;
        var w = (isIframe)? document : window;//parent || window;
		w.removeEventListener('keyup', this.boundKeyUp);
		w.removeEventListener('keydown', this.boundKeyDown);
    	if(bool){
    		w.addEventListener('keyup', this.boundKeyUp, false);
			w.addEventListener('keydown', this.boundKeyDown, false);
            //console.log("isIframe: " + isIframe)
            if(isIframe){
                window.focus();
            }
    	} else {
            this.clear();
        }
    },
    isDown: function(keyCode) {
        return this._pressed[keyCode] > 0;
    },
    wasPressed: function(keyCode) {
        return this._pressed[keyCode] === this.JUST_PRESSED;
    },
    keyState: function(code){
        return this._pressed[code] || this.UP;
    }
};

/**
 * 
 * @constructor
 */
game_shell.KeyListener = function(){};
game_shell.KeyListener.prototype = Object.create(game_shell.Keys);
game_shell.KeyListener.prototype.constructor = game_shell.KeyListener;
//
